### DISCUSSION TITLE
How can i implement a cosine-based similarity?

### DISCUSSION THREAD

**Discussion by chembohuang** [alexklibisz/elastiknn#367] [Q&A]
there are L1, L2, jaccard,, hamming, angular similarity, however, the cosine similarity is missing, r u going to implement one?

**Comment by chembohuang** [comment]
it turns out the angular function similarity is the cosine similarity :)

**Comment by alexklibisz** [comment]
Sorry, just saw this. 
Yes, I'm possibly conflating "angular" and "cosine". The actual implementation is basically what you get if you lookup cosine similarity on Wikipedia.
